### DISCUSSION TITLE
Too many paint transient calls.

### DISCUSSION THREAD

**Discussion by pyscripter** [TurboPack/SynEdit#95] [Ideas]
Various SynEdit operations are surrounded by DoOnPaintTransient or DoOnPaintTransientEx calls.  But it is not consistent.  Other similar operations are not surrounded by such calls and nobody has experienced any issues with that.

At the same time TCustomSynEdit.Paint calls  DoOnPaintTransient but only after doing the painting (ttAfter) and not before.   This is probably what makes things work.   I haven't tested it but I think it would be sufficient to call  DoOnPaintTransient in TCustomSynEdit.Paint both before and after the painting as well as in SetCaretXYEx (as currently done)  and get rid of all other calls to DoOnPaintTransient .

@JaFi-cz Any thoughts?

**Comment by JaFi-cz** [comment]
I use ttAfter. I agree that ttBefore and ttAfter is enough
